src/scenes/Wallet/components/EncryptDialog/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Dialog, { DialogTitle, DialogContent, DialogActions } from 'material-ui/Dialog';
+import { TextField, Button } from 'material-ui';
+import { withStyles } from 'material-ui/styles';
+import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
+import _ from 'lodash';
+
+import styles from './styles';
+import appActions from '../../../../redux/App/actions';
+
+
+const messages = defineMessages({
+  passphrase: {
+    id: 'encrypt.passphrase',
+    defaultMessage: 'Passphrse',
+  },
+});
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+@connect((state) => ({
+  walletEncrypted: state.App.get('walletEncrypted'),
+}), (dispatch) => ({
+  encryptWallet: (passphrase) => dispatch(appActions.encryptWallet(passphrase)),
+}))
+export default class EncryptDialog extends Component {
+  static propTypes = {
+    // classes: PropTypes.object.isRequired,
+    intl: intlShape.isRequired, // eslint-disable-line react/no-typos
+    dialogVisible: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    encryptWallet: PropTypes.func.isRequired,
+    onEncryptWallet: PropTypes.func.isRequired,
+    walletEncrypted: PropTypes.bool.isRequired,
+  };
+
+  static defaultProps = {
+  };
+
+  state = {
+    passphrase: '',
+  };
+
+  render() {
+    const { dialogVisible, onClose, intl, walletEncrypted } = this.props;
+    const { passphrase } = this.state;
+    return (
+      <Dialog
+        open={dialogVisible}
+        onClose={onClose}
+      >
+        <DialogTitle>
+          <FormattedMessage id="wallet.encryptTitle" defaultMessage="Encrypt Your Wallet" />
+        </DialogTitle>
+        <DialogContent>
+          {walletEncrypted ?
+            <FormattedMessage id="wallet.encrypted" defaultMessage="Your wallet is already encrypted" /> :
+            (<span>
+              <FormattedMessage id="wallet.encryptLabel" defaultMessage="Please enter passphrase to encrypt your wallet" />
+              <TextField
+                autoFocus
+                margin="dense"
+                id="toAddress"
+                label={intl.formatMessage(messages.passphrase)}
+                type="string"
+                fullWidth
+                error={_.isEmpty(passphrase)}
+                onChange={this.onPassphraseChange}
```

Comment:
why not just do `onChange={(e) => this.setState({ passphrase: e.target.value })}`? Then we don't have to even make another function.